src/Model/File.php

```diff
@@ -52,10 +74,27 @@ protected function init(): void
         $this->addField('meta_image_width', ['type' => 'integer']);
         $this->addField('meta_image_height', ['type' => 'integer']);
 
-        $this->onHook(Model::HOOK_BEFORE_DELETE, function (self $m) {
-            if ($m->flysystem) { // @phpstan-ignore-line
-                $m->flysystem->delete($m->get('location'));
+        $this->onHook(Model::HOOK_AFTER_DELETE, function (self $m) {
+            $path = $m->get('location');
+            if ($path && $m->flysystem && $m->flysystem->fileExists($path)) { // @phpstan-ignore-line
+                $m->flysystem->delete($path);
             }
         });
     }
+
+    /**
+     * Useful method to clean up all draft files.
+     * Can be called as user action or on schedule bases to clean up filestore repository.
+     *
+     * @return $this
+     */
+    public function cleanupDrafts()
+    {
+        $files = (clone $this->getModel())->addCondition('status', self::STATUS_DRAFT);
```

Comment:
this is not good as there needs to be some delay - otherwise it is subject to cron/user WIP race condition

Reply:
addressed